### DISCUSSION TITLE
Integrating YubiKey as a Second Authentication Factor for SSH via FIDO Tools

### DISCUSSION THREAD

**Discussion by arunwebber** [Yubico/libfido2#830] [Ideas]
I would like to adapt the instructions provided here:

https://github.com/Yubico/libfido2

to the latest version of Tails OS (https://tails.net/)

so that I can enhance the built-in Tails ssh client with fido tools thereby enabling a Yubikey as a second authentication factor to the ssh key pair for logging in to a remote server.

I've got Yubikey for ssh working on Ubuntu, however Tails OS seems to be unable (likely by design) to use the same package management framework and thus produces errors. 

My hope is that by manually assembling a script using dpkg commands, all of the .deb dependencies can be ordered and installed properly. 

The project is essentially to follow the chain of broken dependencies and build the script to successfully install them all so that the Yubikey can be used for ssh on Tails OS.

Unless I'm missing something, there doesn't seem to be official documentation provided by the Tails or Yubico dev teams on this. 

Ideally, this project  would result in an update to their docs to make it easier for anyone in the future to have this feature in Tails. Even more ideally would be for the Tails team to bundle and natively support Yubikey for ssh.

Hopefully a collective effort can make this much needed tooling be more user friendly.

**Comment by LDVG** [comment]
Hi,

Is the ask how to install libfido2 on Tails OS? It is my understanding that it should already be readily available since Tails OS to a large extent relies on the upstream Debian repositories. Any additional support for that would have to be provided by the proper support channels for Tails OS. Let me know if I've misunderstood your questions here.

**Reply by arunwebber** [reply]
Hi its worked My mistake was I was using an old version of tails and once i updated I just run this command. 
` sudo apt install libfido2-1 libfido2-dev libfido2-doc fido2-tools
`
And the tools got installed.
